src/ethereum/index.js

```diff
@@ -25,20 +32,33 @@ export async function connectEth(options = {}) {
 
   let provider
 
-  if (options.isWebsocket) {
-    const websocketURL = env.get('WEB_SOCKET_RPC_URL')
-    if (!websocketURL) {
-      throw new Error(
-        'You need to set the WEB_SOCKET_RPC_URL env var to connect via websockets'
+  switch (options.providerType) {
+    case PROVIDER_TYPES.WEBSOCKET: {
+      const websocketURL = env.get('WEB_SOCKET_RPC_URL')
+      if (!websocketURL) {
+        throw new Error(
+          'You need to set the WEB_SOCKET_RPC_URL env var to connect via websockets'
+        )
+      }
+
+      provider = new providers.WebSocketProvider(websocketURL, {
+        WebSocketConstructor: w3cwebsocket
+      })
+      await provider.connection
+      break
+    }
+    case PROVIDER_TYPES.HDWALLET: {
+      provider = new HDWalletProvider(
+        env.get('HDWALLET_MNEMONIC').replace(/,/g, ' '),
+        env.get('RPC_URL'),
+        env.get('HDWALLET_INDEX', 0)
       )
+      break
+    }
+    case PROVIDER_TYPES.HTTP:
+    default: {
+      provider = env.get('RPC_URL')
     }
```

Comment:
can we add a `break` here? just in case, better to be safe than sorry.

Also you could lose the `{}`